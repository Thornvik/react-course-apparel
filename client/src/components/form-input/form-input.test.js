import React from "react";
import { shallow } from "enzyme";
import { FormInput } from "./form-input.component";

describe("formInput component", () => {
  let wrapper;
  let mockHandleChange;

  beforeEach(() => {
    mockHandleChange = jest.fn();

    const mockProps = {
      handleChange: mockHandleChange,
      label: "email",
      value: "test@gmail.com",
    };

    wrapper = shallow(<FormInput {...mockProps} />);
  });

  it("should render forminput", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should handle change", () => {
    wrapper.find(".form-input").simulate("change");

    expect(mockHandleChange).toHaveBeenCalled();
  });

  it("should render label if label exists", () => {
    expect(wrapper.exists(".form-input-label")).toBe(true);
  });

  it("should not render label if label is null", () => {
    const newMockProps = {
      handleChange: mockHandleChange,
      label: "",
      value: "test@gmail.com",
    };

    const newWrapper = shallow(<FormInput {...newMockProps} />);

    expect(newWrapper.exists(".form-input-label")).toBe(false);
  });
});
