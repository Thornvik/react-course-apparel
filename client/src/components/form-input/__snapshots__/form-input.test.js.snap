// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`formInput component should render forminput 1`] = `ShallowWrapper {}`;
